syncstorage/views/decorators.py

```diff
@@ -192,3 +196,19 @@ def with_collection_lock(viewfunc, request):
 
     with lock_collection(user, collection):
         return viewfunc(request)
+
+
+@make_decorator
+def check_migration(viewfunc, request):
+    """Check if a given user is in a migration state, return 503.
+
+    NOTE: once a user migration has started DO NOT RETURN A 2xx
+    WITHOUT REMOVING THE meta/global RECORD!
+
+    """
+    storage = request.validated["storage"]
+    if storage.is_migrating(request.user):
+        raise HTTPServiceUnavailable(
+            headers={"Retry-After": MIGRATION_RETRY_AFTER}
```

Comment:
Elsewhere in this codebase we use `str` to ensure that header values are strings before returning them, and I have vague recollections of errors that happen if you try to return an int in a header like this. According to #137 this works fine passing it as an integer, but you might like to use `str(MIGRATION_RETRY_AFTER)` here for consistency with other retry-header-setting code.